### PR TITLE
Replace internal serializer with utf8json

### PR DESCRIPTION
**Pull request by russcam**
This PR replaces both the SimpleJson (Elasticsearch.Net) and JSON.Net (in NEST) based internal serializers with a serializer based on [utf8json](https://github.com/neuecc/Utf8Json).

The serializer source has been included in Elasticsearch.Net to allow 
- further modification for the specific need of the client
- mark all serializer types as `internal`

Research as part of #3493 demonstrates an immediate 2x performance improvement in serialization, and it is expected that this could be further improved by

- using `Span<T>`, `Memory<T>` where possible
- reusing `ArrayPool<T>` 
- cleaning and simplifying the interface between serializer and clients, now that the serializer source is included.

All unit tests pass in this PR. One test, `ExtendingNestTypes`, is currently skipped until a general serialization routine is in place. This PR is opened now because there are further changes coming related to 7.x, so the preference is to merge this work in now and fix the remaining failing integration tests which at point of opening are:

```
Failed collections:
 - ReadOnly: Tests.Aggregations.Bucket.Composite.CompositeAggregationUsageTests
 - ReadOnly: Tests.Aggregations.Metric.TopHits.TopHitsAggregationUsageTests
 - ReadOnly: Tests.Search.Request.NestedSortUsageTests
 - ReadOnly: Tests.Search.Request.SearchAfterUsageTests
 - ReadOnly: Tests.Search.Request.SortUsageTests
 - TimeSeries: Tests.XPack.Rollup.RollupJobCrudTests
 - XPack: Tests.XPack.License.GetTrialLicenseStatus.GetTrialLicenseStatusApiTests
 - XPack: Tests.XPack.License.StartTrialLicense.StartTrialLicenseApiTests
 - XPack: Tests.XPack.Migration.MigrationUpgrade.InvalidMigrationUpgradeApiTests
---Reproduce: -----
build.bat seed:51817 integrate 6.4.2 "timeseries,xpack,readonly" "compositeaggregation,tophitsaggregation,nestedsort,searchafter,sort,gettriallicensestatus,starttriallicense,invalidmigrationupgrade,rollupjobcrud"
--------
=== TEST EXECUTION SUMMARY ===
   Tests  Total: 1458, Errors: 0, Failed: 29, Skipped: 13, Time: 689.411s
```

### CODE REVIEW

File: src/CodeGeneration/ApiGenerator/Views/_Requests.Generated.cshtml [hunk 1]
```diff
@@ -13,8 +13,9 @@ using System.Collections.Generic;
 using System.Linq;
 using System.Text;
 using System.Linq.Expressions;
+using System.Runtime.Serialization;
 using Elasticsearch.Net;
-using Newtonsoft.Json;
+using Utf8Json;
```
Comment (russcam) [inline src/CodeGeneration/ApiGenerator/Views/_Requests.Generated.cshtml]: Remove

File: src/Elasticsearch.Net/Utf8Json/Attributes.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Constructor, AllowMultiple = false, Inherited = true)]
+	internal class SerializationConstructorAttribute : Attribute
+    {
+
+    }
+
+    [AttributeUsage(AttributeTargets.Interface, AllowMultiple = false, Inherited = false)]
```
Comment (Mpdreamz) [inline src/Elasticsearch.Net/Utf8Json/Attributes.cs]: A small summary would be good for this one since its used in our domain poco's quite a bit

**Comment by russcam** [comment]
changed base branch from master to 7.x. Will resolve conflicts now

**Comment by Mpdreamz** [comment]
Merged as part of #3608

**Pull request closed without merging**